app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -536,12 +536,18 @@ class PatientRepository @Inject constructor(
   }
 
   fun bpPassportForPatient(patientUuid: UUID): Single<Optional<BusinessId>> {
-    return Single.fromCallable {
-      database
-          .businessIdDao()
-          .latestIdForType(patientUuid, BpPassport)
-          .toOptional()
-    }
+    return database
+        .businessIdDao()
+        .latestIdForType(patientUuid, BpPassport)
+        .toObservable()
+        .map { bpPassports ->
+          if (bpPassports.isEmpty()) {
+            None
+          } else {
+            bpPassports.first().toOptional()
+          }
+        }
+        .firstOrError()
```

Comment:
This method can return an `Observable<Optional<BusinessId>>` and then consumers can decide whether they need a `Single` or not. I personally prefer this approach, because it also allows the same method to publish change notifications downstream. What do you think?